sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -276,44 +276,53 @@ class ResolveSessionCatalog(
     case c @ CreateTableStatement(
          SessionCatalogAndTable(catalog, tbl), _, _, _, _, _, _, _, _, _, _, _) =>
       assertNoNullTypeInSchema(c.tableSchema)
-      buildV1Table(tbl.asTableIdentifier, c) match {
-        case Some(tableDesc) =>
-          val mode = if (c.ifNotExists) SaveMode.Ignore else SaveMode.ErrorIfExists
-          CreateTable(tableDesc, mode, None)
-
-        case None =>
+      val (storageFormat, provider) = getStorageFormatAndProvider(
```

Comment:
I did some refactoring, to avoid duplicated code between CREATE TABLE and CTAS, `buildCatalogTable` and `buildHiveCatalogTable`

Reply:
Looks fine to me. I like that it should have fewer changes from master.